pages/docs/guides/rate-limiting.mdx

```diff
@@ -0,0 +1,103 @@
+export const description = 'Prevent excessive function runs over a given time period by skipping events beyond a specific limit. Ideal for protecting against abuse.'
+
+# Rate limiting
+
+Rate limiting is a _hard limit_ on how many function runs can start within a time period. Events that exceed the rate limit are _skipped_ and do not trigger functions to start. This prevents excessive function runs over a given time period. Some use cases for rate limiting include:
+
+* Preventing abuse of your system
+* Reducing frequency of data synchronization functions
+* Smoothing out noisy [webhook](/docs/platform/webhooks) events
+
+## How to configure rate limiting
+
+<CodeGroup forceTabs>
+```ts
+export default inngest.createFunction(
+  {
+    id: "synchronize-data",
+    rateLimit: {
+      limit: 1,
+      period: "4h",
+      key: "event.data.company_id",
+    },
+  },
+  { event: "intercom/company.updated" },
+  async ({ event, step }) => {
+    // This function will be rate limited
+    // It will only run 1 once per 4 hours for a given event payload with matching company_id
+  }
+);
+```
+</CodeGroup>
+
+### Configuration reference
+
+* `limit` - The maximum number of functions to run in the given time period.
+* `period` - The time period of which to set the limit. The period begins when the first matching event is received.
```

Comment:
"The time period of which to set the limit." doesn't sound intuitive to me but I don't have a suggestion atm on how to make it easy.